nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-framework-core/src/main/java/org/apache/nifi/controller/reporting/AbstractReportingTaskNode.java

```diff
@@ -125,7 +141,12 @@ public void setSchedulingPeriod(final String schedulingPeriod) {
 
     @Override
     public ReportingTask getReportingTask() {
-        return reportingTask;
+        return reportingTaskRef.get().getReportingTask();
+    }
+
+    @Override
+    public void setReportingTask(final LoggableComponent<ReportingTask> reportingTask) {
+        this.reportingTaskRef.set(new ReportingTaskDetails(reportingTask));
```

Comment:
We should make sure here that the Reporting Task is stopped, similar to Processors. And should also ensure that Controller Services are disabled when calling setControllerService.